multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -306,6 +306,10 @@ def _fit_x(self, X: pd.DataFrame):
         if self._fit_x_called:
             raise RuntimeError("fit_x() has been called. Please create a new preprocessor and call it again!")
         self._fit_x_called = True
+        # Creating deep copy of the DataFrame, which allows writable buffer to be created for the new df
+        # This is needed for 1.4.1 < scikit-learn < 1.5.0, the versions 1.4.0 and 1.5.1 do not need a writable buffer
```

Comment:
If this is a bug, then it would be good to link to a sklearn GitHub issue

Reply:
https://github.com/scikit-learn/scikit-learn/issues/29182

updated the PR comment